src/node/txreconciliation.cpp

```diff
@@ -124,6 +132,33 @@ class TxReconciliationTracker::Impl
         return ReconciliationRegisterResult::SUCCESS;
     }
 
+    void AddToSet(NodeId peer_id, const std::vector<uint256>& txs_to_reconcile) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        Assume(txs_to_reconcile.size() > 0);
+        LOCK(m_txreconciliation_mutex);
+        assert(IsPeerRegistered(peer_id));
+        auto& recon_state = std::get<TxReconciliationState>(m_states.find(peer_id)->second);
+
+        size_t added = 0;
+        for (auto& wtxid: txs_to_reconcile) {
+            added += recon_state.m_local_set.insert(wtxid).second;
+        }
+
+        LogPrint(BCLog::NET, "Added %i new transactions to the reconciliation set for peer=%d. " /* Continued */
+            "Now the set contains %i transactions.\n", added, peer_id, recon_state.m_local_set.size());
+    }
+
+    void TryRemovingFromSet(NodeId peer_id, const uint256& wtxid_to_remove) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        assert(IsPeerRegistered(peer_id));
+        auto& recon_state = std::get<TxReconciliationState>(m_states.find(peer_id)->second);
+
+        recon_state.m_local_set.erase(wtxid_to_remove);
```

Comment:
The removal is currently per-tx (not batched), so it would make the logs really annoying... Which raises another question: should I refactor it to make it batched where possible (on hearing INV batch from the given peer)?